akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupBySpec.scala

```diff
@@ -552,24 +553,19 @@ class FlowGroupBySpec extends StreamSpec("""
     }
 
     "work with random demand" in assertAllStagesStopped {
-      var blockingNextElement: ByteString = null.asInstanceOf[ByteString]
-
-      val probes = new java.util.ArrayList[Promise[TestSubscriber.Probe[ByteString]]](100)
-      (0 to 99).foreach(_ => probes.add(Promise[TestSubscriber.Probe[ByteString]]()))
-
-      var probesWriterTop = 0
-      var probesReaderTop = 0
-
-      case class SubFlowState(probe: TestSubscriber.Probe[ByteString], hasDemand: Boolean, firstElement: ByteString)
-      val map = new util.HashMap[Int, SubFlowState]()
+      val probes = IndexedSeq.tabulate(100)(_ => Promise[TestSubscriber.Probe[ByteString]]())
```

Comment:
Not sure why the test was using Java collections throughout, changed to Scala counterparts.